zio-http/src/main/scala/zio/http/Http.scala

```diff
@@ -721,7 +725,8 @@ object Http {
   /**
    * Provides access to the request's ChannelHandlerContext
    */
-  def context: Http[Any, Nothing, Request, ChannelHandlerContext] = Http.fromFunction[Request](_.unsafeContext)
+  def context: Http[Any, Nothing, Request, ChannelHandlerContext] =
+    Http.fromFunctionZIO[Request](request => ZIO.succeedUnsafe { implicit u => request.unsafe.context })
```

Comment:
To avoid the extra allocation, maybe just `Http.fromFunction { request => request.unsafe.context(Unsafe.unsafe) }`